### PR TITLE
Resolve SonarCloud Bugs

### PR DESCRIPTION
**Pull request by naseberry**
## What

[SonarCloud - Apply - Bugs](https://dsdmoj.atlassian.net/browse/AP-3520)

Resolve SonarClouds reported bugs. This resolves some of the existing issues in order to reduce and resolve our quality gate thresholds.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: app/views/layouts/pdf.html.erb [hunk 1]
```diff
@@ -2,6 +2,7 @@
 <html lang="en" class="govuk-template">
   <head>
     <meta charset="utf-8">
+    <title><%= t("layouts.application.header.title") %></title>
```
Comment (colinbruce) [inline app/views/layouts/pdf.html.erb:5]: I'd like to UAT this and ensure it isn't doing something weird with the report generator! :+1:

File: app/views/saml_idp/idp/saml_post.html.erb [hunk 2]
```diff
@@ -1,8 +1,9 @@
 <!DOCTYPE html>
-<html>
+<html lang="en">
   <head>
     <meta charset='utf-8'>
     <meta http-equiv='X-UA-Compatible' content='IE=edge,chrome=1'>
+    <title><%= t("layouts.application.header.title") %></title>
```
Comment (colinbruce) [inline app/views/saml_idp/idp/saml_post.html.erb:6]: Ditto portal login

**Review by colinbruce — APPROVED** [review]
Looks good, but a couple of UAT tests identified :+1:

**Review by jsugarman — APPROVED** [review]
looks good, baring Colin's UAT comments on pdf generation.

File: app/views/citizens/legal_aid_applications/index.html.erb [hunk 3]
```diff
@@ -1,13 +1,13 @@
 <%= page_template page_title: t('.heading_1'), back_link: :none do %>
   <p class="govuk-body">
-    <b> <%= t('.name') %></b>
+    <strong> <%= t('.name') %></strong>
```
Comment (cpjmcquillan) [inline app/views/citizens/legal_aid_applications/index.html.erb:3]: I have a couple questions about the use of `<strong>` tags.

I'm intrigued what the justification is from SonarCloud for these being "Code smells"?

The [`<b>` tag is a valid HTML element with its own purpose](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/b).

Do we want to communicate to the user that all of these elements are important or urgent?

```html
This is some text that is <strong>seriously important!</strong>
```

Or maybe we want to draw attention to elements without indicating that they are especially important?

```html
This is <b>some text that isn't important</b>, but I want to draw attention to it.
```

Maybe we just want to style some text with a bold weight?

```html
This is <span class="govuk-!-font-weight-bold">some text I want to look bold</span>
```

I think it's important we are conscious about the the tags we use, and we should be writing semantic HTML wherever we can. 

I'm worried SonarCloud just says "\<strong> tag good" and we follow it blindly without proper consideration.

What do you think?

File: app/views/saml_idp/idp/saml_post.html.erb (hunk 2 above)
Comment (naseberry) [reply app/views/saml_idp/idp/saml_post.html.erb:6]: > Ditto portal login

Great. I decided to use the service name in these instances as we weren't yielding any page title values to the layout page.

**Comment by colinbruce** [comment]
UAT'd the PDF and it seems okay :+1:
Not sure we can do much with Portal login until it's on staging.  
@naseberry shout when you are deploying and I can UAT the staging login before you deploy to prod :+1:

**Pull request merged**